libs/server/Resp/RespServerSession.cs

```diff
@@ -353,6 +369,29 @@ private void ProcessMessages()
                 // Check ACL permissions for the command
                 if (cmd != RespCommand.INVALID && CheckACLPermissions(cmd))
                 {
+
+                    if (runningWithAOFWaitForCommitMode)
```

Comment:
This is polluting the main ProcessMessages. Can it be moved to CheckACLPermissions and tighten the long comment.

Reply:
can be moved, but not completely sure by what you mean by tightening the comment. Do you just want me to get rid of the whitespaces between paragraphs?

Reply:
do we really need the full truth table there? seems implicit from the boolean logic ... i am on the fence w.r.t. this so if you think it adds significant value, it can be kept. just a small nit/preference.

Reply:
I usually tend to leave the work in if something took me longer than 2 minutes to come up with. Although looking back, I think we can get rid of the truth table since the comment above tells you they "why" pretty well.

Reply:
removed